telephone-line-segments.el

```diff
@@ -105,9 +122,10 @@ Adapted from doom-modeline."
   buffer-file-name)
 
 (telephone-line-defsegment* telephone-line-buffer-modified-segment ()
-    (if (buffer-modified-p)
-        (telephone-line-raw "!")
-      (telephone-line-raw "-")))
+  (cond
+   (buffer-read-only "·")
```

Comment:
While it's not as clean, I think it might be better to stick to `%` for this since that's what the builtin modeline does.